WordPress/gutenberg
# [Mobile] - Enable font size and line height support and addresses Android issues

- Gutenberg mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4519
- Aztec Android PR: https://github.com/wordpress-mobile/AztecEditor-Android/pull/952
- WordPress Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/15868
- WordPress iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/17833

## Description
This PR enables font size and line-height support on mobile for block-based themes.

It also includes some fixes on Android and the Heading block.

On Aztec Android, different scales are added to the Heading block. Unfortunately, I couldn't find an easy way to disable this for an specific case like block-based themes. So for now the approach is the following:

- [Bring the scales](https://github.com/WordPress/gutenberg/pull/38205/files#diff-18c4abf20356408efc28d1d949b45f1b8f479898f849eb0af84e538ebd9322f4R282) used in [Aztec](https://github.com/wordpress-mobile/AztecEditor-Android/blob/trunk/aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecHeadingSpan.kt#L95-L100) to [subtract it](https://github.com/WordPress/gutenberg/pull/38205/files#diff-18c4abf20356408efc28d1d949b45f1b8f479898f849eb0af84e538ebd9322f4R331) from the font size before setting the value.
- It adds `isHeadingBlock` to check if the current tag is from a Heading block, e.g `h1`, `h2`, etc.
- Whenever `setBlockType` is called, it triggets `setFontSize`, for cases when the Heading block was already created but its level changes e.g from `h1` to `h2`.

This also limits the minimum value that can be set for line-height to `1`, lower values break on Android where the block doesn't show the correct height.

## Testing Instructions

## Font size (block-based themes: Calvin, Quadrat, etc.)

### Test case 1

- Add a Paragraph block
- Type in some text
- Open the block settings
- Change the font size
- **Expect** to see the text with the selected new size

### Test case 2

- Add a Heading block
- Type in some text
- Change through different levels, `h1`, `h3`, `h4`
- **Expect** to see the block change accordingly

### Test case 3

- Add a Heading block
- Set the `H1` level
- Type in a long text so there are multiple lines
- **Expect** the text to show correctly without overlapping
- Open the block settings
- Change the font size
- **Expect** to see the text with the new font size
- Change to the `H4` level
- **Expect** the text to keep the same font size selected before
- Open the block settings
- Select the `Default` font size
- **Expect** to see the font size go back to the selected heading level default size

## Line-height (block-based themes: Calvin, Quadrat, etc.)

### Test case 1

- Open the app
- Add a Paragraph block
- Type in several lines of text
- Open the block settings
- Change the line-height
- **Expect** to see the line height to change

### Test case 2

- Add a Heading block
- Type in several lines of text
- Open the block settings
- Change the line-height
- **Expect** to see the line height to change
- Change the heading level to `H1`
- **Expect** the font size value to change and the line-height to update accordingly to the new size

### Test case 3

- Add a Heading block
- Type in several lines of text
- Open the block settings
- **Check** that you **can't select** a line-height value lower than `1`

## Font size (standard theme: Seedlet)

### Test case 1

- Open the app
- Add a Paragraph block
- Type in some text
- Open the block settings
- **Expect** to **not** see the font size picker

### Test case 2

- Open the app
- Add a Heading block
- Type in some text
- Change to `H1`
- **Expect** the font size to change
- Change to `H3`
- **Expect** the font size to change

## Line-height (standard theme: Seedlet)

### Test case 1

- Open the app
- Add a Heading block
- Type in some text
- Open the block settings
- **Expect** to **not** see the line height picker

## Screenshots 

## Font size (block-based themes: Calvin, Quadrat, etc.)

Test case 1 | Test case 2 | Test case 3
-|-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/151434659-8f43cb47-9eb6-4ca7-9cbb-07d6ee8ca77a.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/151434667-397813b4-ac11-4bb6-9760-a4c444549f8a.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/151434672-8c6ef7cc-9e70-41c5-b2f2-660cfc7c73f5.gif" width="200" /></kbd>

## Line-height (block-based themes: Calvin, Quadrat, etc.)

Test case 1 | Test case 2 | Test case 3
-|-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/151435088-8fe0a0a7-884e-40cf-85e9-187ac341e0c4.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/151435094-f5e76c28-25e4-4be4-8044-3096b73eaf59.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/151435103-e628a3a0-082a-409d-ac7e-2ca592079b95.gif" width="200" /></kbd>

## Font size (standard theme: Seedlet)

Test case 1 | Test case 2
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/151435355-e78ad044-237c-4cd3-ab3e-5fba745178e2.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/151435364-9cd98db7-14af-4481-9500-58963da19dbc.gif" width="200" /></kbd>

## Line-height (standard theme: Seedlet)

Test case 1
<kbd><img src="https://user-images.githubusercontent.com/4885740/151435533-528b5c22-018b-4bf3-85eb-7c1da77e6924.gif" width="200" /></kbd>

## Types of changes
New feature

## Checklist:
